src/main/java/laddergame/domain/gameelements/Prizes.java

```diff
@@ -1,16 +1,30 @@
 package laddergame.domain.gameelements;
 
+import java.util.ArrayList;
 import java.util.List;
 
 public class Prizes {
 
-    private final List<Name> prizeNames;
+    private final List<Prize> prizes;
 
     public Prizes(List<String> prizeNames, int playerNumber) {
         validateSameLength(prizeNames.size(), playerNumber);
-        this.prizeNames = prizeNames.stream()
-                .map(Name::new)
-                .toList();
+
+        this.prizes = new ArrayList<>();
+
+        for (int i = 0; i < prizeNames.size(); i++) {
+            Name prizeName = new Name(prizeNames.get(i));
+            Position position = new Position(i);
+
+            this.prizes.add(new Prize(prizeName, position));
+        }
+    }
+
+    public Prize findSamePositionPrize(Position playerPosition) {
+        return prizes.stream()
+                .filter(prize -> prize.getPosition().isSame(playerPosition))
```

Comment:
position 비교에 대한 검증도 prize에게 넘겨볼 수 있겠네요
```suggestion
                .filter(prize -> prize.isSamePosition(playerPosition))
```

Reply:
비교 책임 이전에 대한 반영이 안되어 있었네요!! 옮기니 훨씬 깔끔해지고 객체간 역할 분배도 더 명확해진 것 같습니다!